Include/internal/pycore_interp_structs.h

```diff
@@ -757,6 +757,30 @@ struct _Py_unique_id_pool {
 
 typedef _Py_CODEUNIT *(*_PyJitEntryFuncPtr)(struct _PyExecutorObject *exec, _PyInterpreterFrame *frame, _PyStackRef *stack_pointer, PyThreadState *tstate);
 
+typedef struct _PyJitTracerState {
+    bool dependencies_still_valid;
+    bool prev_instr_is_super;
+    int code_max_size;
+    int code_curr_size;
+    int initial_stack_depth;
+    int initial_chain_depth;
+    int prev_instr_oparg;
+    int prev_instr_stacklevel;
+    int specialize_counter;
+    _PyUOpInstruction *code_buffer;
+    _Py_CODEUNIT *insert_exec_instr;
+    _Py_CODEUNIT *close_loop_instr;
```

Comment:
The code involving this and `insert_exec_instr` seems complex and its not clear what the heuristics are for stopping tracing. Maybe just stop on the second backwards jump if it doesn't rejoin the start?